src/server/pfs/server/api_server.go

```diff
@@ -272,13 +272,33 @@ func (a *apiServer) GetFile(request *pfs.GetFileRequest, apiGetFileServer pfs.AP
 	func() { a.Log(request, nil, nil, 0) }()
 	defer func(start time.Time) { a.Log(request, nil, retErr, time.Since(start)) }(time.Now())
 
-	file, err := a.driver.getFile(a.getPachClient(apiGetFileServer.Context()), request.File, request.OffsetBytes, request.SizeBytes)
+	file, _, err := a.driver.getFile(a.getPachClient(apiGetFileServer.Context()), request.File, request.OffsetBytes, request.SizeBytes)
 	if err != nil {
 		return err
 	}
 	return grpcutil.WriteToStreamingBytesServer(file, apiGetFileServer)
 }
 
+func (a *apiServer) GetFiles(request *pfs.GetFileRequest, apiGetFilesServer pfs.API_GetFilesServer) (retErr error) {
```

Comment:
I think the biggest open question I have currently is how we want to structure the driver functions that `GetFiles()` calls. I have currently taken a naive approach to where I just modify `driver.getFile()` to now also return a slice of  `NodeProto`s that is then used to parse the stream of bytes into individual files.

The assumption I am making is that the size of this is small enough to keep in memory and we can just iterate over it. Let me know if this assumption is false. 

This will most likely all be re-worked for the next step which is to enable calling `GetFiles` on directories, so I did not want to spent too much time building out something here that would later be scrapped.

Reply:
What makes sense to me, having not thought about this as much as you have, is to add a new method `getFiles` that looks something like this:

```go
func (d *driver) getFiles(pachClient *client.APIClient, file *pfs.File, func((*pfs.File, io.Reader) error) error {
    ...
}
```

As I've mentioned before I really like callback style iteration in Go, but I think it gets really powerful when you have it at every layer because it's very easy to compose.